router/xgress_edge_tunnel/fabric.go

```diff
@@ -363,8 +365,7 @@ func (self *fabricProvider) establishTerminator(terminator *tunnelTerminator) er
 		return err
 	}
 
-	time.Sleep(250 * time.Millisecond)
-	if terminator.created.Load() {
+	if terminator.WaitForCreated(10 * time.Second) {
 		return nil
 	}
 
```

Comment:
since i can't comment on line 373 (below)... I'll comment here. 

```
return errors.New("check back later")
```
probably isn't super duper useful if to me if i ended up seeing it?

Reply:
I'll change this to something like: current attempt timed out, with some context info

Reply:
It's generally not seen, it's mostly a signal to the retry mechanism. It can show up in some cases, though.

Reply:
Fixed